CoreWidgetProvider/Helpers/ChartHelper.cs

```diff
@@ -1,138 +1,226 @@
+    }
+
+    private static XElement CreateBlankSvg(int height, int width)
+    {
+        var svgElement = new XElement("svg");
+        svgElement.SetAttributeValue("height", height);
+        svgElement.SetAttributeValue("width", width);
+        return svgElement;
+    }
+
+    private static XElement CreateGradientDefinition(ChartType type)
+    {
+        var defsElement = new XElement("defs");
+        var gradientElement = new XElement("linearGradient");
+
+        // Vertical gradients are created when x1 and x2 are equal and y1 and y2 differ.
+        gradientElement.SetAttributeValue("x1", "0%");
+        gradientElement.SetAttributeValue("x2", "0%");
+        gradientElement.SetAttributeValue("y1", "0%");
+        gradientElement.SetAttributeValue("y2", "100%");
+        gradientElement.SetAttributeValue("id", "gradientId");
+
+        string stop1Style;
+        string stop2Style;
+        switch (type)
+        {
+            case ChartType.GPU:
+                stop1Style = GPUBrushStop1Style;
+                stop2Style = GPUBrushStop2Style;
+                break;
+            case ChartType.Mem:
+                stop1Style = MemBrushStop1Style;
+                stop2Style = MemBrushStop2Style;
+                break;
+            case ChartType.Net:
+                stop1Style = NetBrushStop1Style;
+                stop2Style = NetBrushStop2Style;
+                break;
+            case ChartType.CPU:
+            default:
+                stop1Style = CPUBrushStop1Style;
+                stop2Style = CPUBrushStop2Style;
+                break;
+        }
+
+        var stop1 = new XElement("stop");
+        stop1.SetAttributeValue("offset", "0%");
+        stop1.SetAttributeValue("style", stop1Style);
+
+        var stop2 = new XElement("stop");
+        stop2.SetAttributeValue("offset", "95%");
+        stop2.SetAttributeValue("style", stop2Style);
+
+        gradientElement.Add(stop1);
+        gradientElement.Add(stop2);
+        defsElement.Add(gradientElement);
+
+        return defsElement;
+    }
+
+    private static XElement CreateBorderBox(int height, int width)
+    {
+        var boxElement = new XElement("rect");
+        boxElement.SetAttributeValue("height", height);
+        boxElement.SetAttributeValue("width", width);
+        boxElement.SetAttributeValue("style", LightGrayBoxStyle);
+        return boxElement;
+    }
+
+    private static string GetLineStyle(ChartType type)
+    {
+        var lineStyle = type switch
+        {
+            ChartType.CPU => CPULineStyle,
+            ChartType.GPU => GPULineStyle,
+            ChartType.Mem => MemLineStyle,
+            ChartType.Net => NetLineStyle,
+            _ => CPULineStyle,
+        };
+
+        return lineStyle;
+    }
+
+    private static StringBuilder TransformPointsToLine(List<float> chartValues, out int startX, out int finalX)
+    {
+        var points = new StringBuilder();
+
+        var pxBtwnPoints = 9;

```

Comment:
is 9 derived from something else?

Reply:
Just that it fits the 30 points in the space the best. Moved this line under the comment below it, where it explains.